web_security_dialog/models/res_company.py

```diff
@@ -0,0 +1,20 @@
+# -*- coding: utf-8 -*-
+# See LICENSE file for full copyright and licensing details.
+from odoo import api, fields, models
+
+
+class ResCompany(models.Model):
+    _inherit = 'res.company'
+
+    security_key = fields.Char('Security Code')
+
+    @api.multi
+    def check_security(self, vals):
+        fields = vals.get('field').encode('ascii', 'ignore')
+        result = self.search_read([('id', '=', vals.get('companyId'))],
```

Comment:
you don't need to search when you already have the id. Use `self.browse`

Reply:
ah, and don't forget to write a test for this function to satisfy the codecov CI

Reply:
Done